pizza_cutter/slice_utils/interpolate.py

```diff
@@ -74,7 +77,128 @@ def _interp_patch(*, image, bad_msk, i, j, size, buff):
         return None
 
 
+@njit
+def _get_nearby_good_pixels(image, bad_msk, buff=3):
+    """
+    get the set of good pixels surrounding bad pixels.
+
+    Parameters
+    ----------
+    image: array
+        The image data
+    bad_msk: bool array
+        2d array of mask bits.  True means it is a bad
+        pixel
+
+    Returns
+    -------
+    bad_pix:
+        bad pix is the set of bad pixels, shape [nbad, 2]
+    good_pix:
+        good pix is the set of bood pixels around the bad
+        pixels, shape [ngood, 2]
+    good_im:
+        the set of good image values, shape [ngood]
+    good_ind:
+        the 1d indices of the good pixels row*ncol + col
+    """
+
+    nrows, ncols = bad_msk.shape
+
+    ngood = bad_msk.size
+    good_pix = np.zeros((ngood, 2), dtype=numba.int64)
+    good_ind = np.zeros(ngood, dtype=numba.int64)
+    bad_pix = np.zeros((ngood, 2), dtype=numba.int64)
+    good_im = np.zeros(ngood, dtype=image.dtype)
+
+    ibad = 0
+    igood = 0
+    for row in range(nrows):
+        for col in range(ncols):
+            val = bad_msk[row, col]
+            if val:
+                bad_pix[ibad] = (row, col)
+                ibad += 1
+
+                row_start = row - buff
+                row_end = row + buff
+                col_start = col - buff
+                col_end = col + buff
+                for rc in range(row_start, row_end+1):
+                    if rc == row:
+                        continue
+                    for cc in range(col_start, col_end+1):
+                        if cc == col:
+                            continue
+
+                        tval = bad_msk[rc, cc]
+                        if not tval:
+
+                            if igood == ngood:
+                                raise RuntimeError('good_pix too small')
+
+                            # got a good one, add it to the list
+                            good_pix[igood] = (rc, cc)
+                            good_im[igood] = image[rc, cc]
+
+                            # keep track of index
+                            ind = rc*ncols + cc
+                            good_ind[igood] = ind
+                            igood += 1
+
+    bad_pix = bad_pix[:ibad, :]
+
+    good_pix = good_pix[:igood, :]
+    good_ind = good_ind[:igood]
+    good_im = good_im[:igood]
+
+    return bad_pix, good_pix, good_im, good_ind
+
+
 def _grid_interp(*, image, bad_msk):
+    """
+    interpolate the bad pixels in an image
+
+    Parameters
+    ----------
+    image: array
+        the pixel data
+    bad_msk: array
+        boolean array, True means it is a bad pixel
+    """
+
+    nrows, ncols = image.shape
+    npix = bad_msk.size
+
+    bad_pix, good_pix, good_im, good_ind = \
+        _get_nearby_good_pixels(image, bad_msk)
+
+    nbad = bad_pix.shape[0]
+    bm_frac = nbad/npix
+
+    if bm_frac < 0.90 and npix - nbad > 10:
```

Comment:
@beckermr is this `npix - nbad > 10` still the right check now that we are doing the image all at once?

Reply:
nope - probably not needed anymore

Reply:
removed